main/src/ca/uwaterloo/flix/language/errors/EntryPointError.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2022 Matthew Lutze
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.language.errors
+
+import ca.uwaterloo.flix.language.CompilationMessage
+import ca.uwaterloo.flix.language.ast.{SourceLocation, Symbol, Type}
+import ca.uwaterloo.flix.language.fmt.{Audience, FormatType}
+import ca.uwaterloo.flix.util.Formatter
+
+sealed trait EntryPointError extends CompilationMessage {
+  val kind: String = "Entry Point Error"
+}
+
+object EntryPointError {
+
+  private implicit val audience: Audience = Audience.External
+
+  /**
+    * Error indicating an unexpected argument type to an entry point function.
+    *
+    * @param sym the entry point function.
+    * @param tpe the argument type.
+    * @param loc the location where the error occurred.
+    */
+  case class UnexpectedEntryPointArg(sym: Symbol.DefnSym, tpe: Type, loc: SourceLocation) extends EntryPointError {
+
+    override def summary: String = s"Unexpected entry point argument type: ${FormatType.formatWellKindedType(tpe)}."
+
+    override def message(formatter: Formatter): String = {
+      import formatter._
+      s"""${line(kind, source.name)}
+         |>> The argument type: '${red(FormatType.formatWellKindedType(tpe))}' is not a valid entry point argument type.
```

Comment:
In the interest of both clarity and brevity. How about instead simply:

`Expected Unit or Array[String], but found: ...`

Reply:
This would apply to the summary and both places in the message.